path_config.go

```diff
@@ -266,6 +280,12 @@ func (b *jwtAuthBackend) pathConfigWrite(ctx context.Context, req *logical.Reque
 		return logical.ErrorResponse("invalid response_mode: %q", config.OIDCResponseMode), nil
 	}
 
+	// Validate provider_config
+	_, err := NewProviderConfig(config, ProviderMap)
```

Comment:
Do we anticipate developers constructing their own `ProviderMap` types and using them in `NewProviderConfig`? If we plan on only supporting those defined in the package var `ProviderMap` then perhaps `NewProviderConfig` should just use that directly and not accept it as input?

Reply:
I was thinking there would be one `ProviderMap` where all the custom types are mapped to their names. The reason `ProviderMap` is an input to `NewProviderConfig()` is to make it easier to test the error handling in `NewProviderConfig()`.

Reply:
👍